cray_product_catalog/catalog_delete.py

```diff
@@ -219,7 +219,29 @@ def modify(self):
             return
 
         self.__modify_main_cm()
-        self.__modify_product_cm()
+
+        """Invoke __modify_product_cm function only if the product_cm is present
+
+        Before attempting to delete a configmap, first verify if for the product there is a product_cm present or not.
+        If not, give a warning message and continue
+        """
```

Comment:
Triple quotes should not be used for regular code comments (function/method/class/module docstrings are an exception).
```suggestion
        # Invoke __modify_product_cm function only if the product_cm is present

        # Before attempting to delete a ConfigMap, first verify if for the product there is a product_cm present or not.
        # If not, give a warning message and continue
```